flexget/plugins/parsers/parser_guessit.py

```diff
@@ -363,11 +169,178 @@ def parse_series(self, data, **kwargs):
         except GuessitException:
             log.warning('Parsing %s with guessit failed. Most likely a unicode error.', data)
             guess_result = {}
-        parsed = GuessitParsedSerie(data, kwargs.pop('name', None), guess_result, **kwargs)
+
+        if guess_result.get('type') != 'episode':
+            valid = False
+
+        name = kwargs.get('name')
+        country = guess_result.get('country')
+        if not name:
+            name = guess_result.get('title')
+            if country and hasattr(country, 'alpha2'):
+                name += ' (%s)' % country.alpha2
+        else:
+            # Make sure the name match is up to FlexGet standards
+            # Check there is no unmatched cruft before the matched name
+            title_start = guess_result.matches['title'][0].start
+            title_end = guess_result.matches['title'][0].end
+            if title_start != 0:
+                try:
+                    pre_title = max((match[0].end for match in guess_result.matches.values() if
+                                     match[0].end <= title_start))
+                except ValueError:
+                    pre_title = 0
+                for char in reversed(data[pre_title:title_start]):
+                    if char.isalnum() or char.isdigit():
+                        return SeriesParseResult(data=data, valid=False)
+                    if char.isspace() or char in '._':
```

Comment:
i know there's nothing wrong in iterating over a string like that but it looks kinda weird to me

Reply:
Pretty normal when parsing tokens.